packages/material-ui/src/utils/unstable_useId.js

```diff
@@ -0,0 +1,18 @@
+import * as React from 'react';
+
+/**
+ * Private module reserved for @material-ui/x packages.
+ */
+export default function useId(idOverride) {
+  const [defaultId, setDefaultId] = React.useState(idOverride);
+  const id = idOverride || defaultId;
```

Comment:
Just curious and the answer might be obvious - but why wouldn’t we just use the value in state?

Reply:
Good question and I think perhaps if I split out the second assertion from the tests, it would've been clearer. It's to support updating `idOverride`, e.g. if for whatever reason they decided to change the `id` prop.